### PR TITLE
[FEATURE] Add methods to fetch collection(s) with products

### PR DESCRIPTION
**Pull request by jzjiang**
Add two methods:
1. `fetchAllCollectionsWithProducts`
2. `fetchCollectionWithProducts`

Images and variants will be paginated on the products returned.

These methods don't have the option to specify fields to query to avoid confusion (e.g. if you start specifying `products` on `fetchCollection` or create a query without `products` and send it to `fetchAllCollectionsWithProducts`)

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -157,6 +157,36 @@ export default class Client {
...
+            product.attrs.images = images;
+          }));
+
+          promiseAcc.push(this.graphQLClient.fetchAllPages(product.variants, {pageSize: 250}).then((variants) => {
+            product.attrs.variants = variants;
+          }));
+
+          return promiseAcc;
+        }, []);
+      }, []);
```
Comment (minasmart) [inline src/client.js]: This probably shouldn't fetch all products. Fetch the first set, and we expose pagination on the client, so let people paginate. I'd limit this to something much smaller like 20? But at least make it a param on the method.

File: src/client.js [hunk 2]
```diff
@@ -167,6 +197,32 @@ export default class Client {
...
+        promiseAcc.push(this.graphQLClient.fetchAllPages(product.images, {pageSize: 250}).then((images) => {
+          product.attrs.images = images;
+        }));
+
+        promiseAcc.push(this.graphQLClient.fetchAllPages(product.variants, {pageSize: 250}).then((variants) => {
+          product.attrs.variants = variants;
+        }));
+
+        return promiseAcc;
+      }, []);
```
Comment (minasmart) [inline src/client.js]: Same comment as above.

File: src/client.js (hunk 1 above)
Comment (jzjiang) [inline src/client.js]: It fetches images/variants on each product in the collections, not all the products.

Comment (minasmart) [inline src/client.js]: ok, maybe share this behaviour than? Since it's now used the same way in I think 3 places.

**Comment by jzjiang** [comment]
Since @mina is gone for the week, could you review @lshapton?

File: src/collection-node-query.js [hunk 3]
```diff
@@ -1,8 +1,13 @@
...
 
 export const defaultFields = ['id', 'handle', 'updatedAt', 'title', ['image', imageQuery()]];
 
 export default function collectionNodeQuery(fields = defaultFields) {
   return nodeQuery('Collection', fields);
 }
+
+export function collectionNodeWithProductsQuery() {
+  return nodeQuery('Collection', [...defaultFields, ['products', productConnectionQuery()]]);
+}
```
Comment (lshapton) [inline src/collection-node-query.js]: Where is this used? I see `collectionNodeQuery ` is used in `client.js`

Comment (jzjiang) [inline src/collection-node-query.js]: I forgot to remove this. Thanks for the catch! :)

**Pull request merged**